internal/provider/camunda_cluster_connector_secret_resource.go

```diff
@@ -0,0 +1,222 @@
+	// Provider not yet configured
+	if req.ProviderData == nil {
+		return
+	}
+
+	provider, ok := req.ProviderData.(*CamundaCloudProvider)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *CamundaCloud client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.provider = provider
+}
+
+func (r *CamundaClusterConnectorSecretResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data camundaClusterConnectorSecret
+
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	newClusterConnectorSecretConfiguration := console.CreateSecretBody{
+		SecretName:  data.Name.ValueString(),
+		SecretValue: data.Value.ValueString(),
+	}
+
+	ctx = context.WithValue(ctx, console.ContextAccessToken, r.provider.accessToken)
+
+	inline, err := r.provider.client.ClustersApi.
+		CreateSecret(ctx, data.ClusterId.ValueString()).
+		CreateSecretBody(newClusterConnectorSecretConfiguration).
+		Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create cluster connector secret",
+			fmt.Sprintf("Unable to create cluster connector secret, got error: %s",
+				err.(*console.GenericOpenAPIError).Body()),
+		)
+		return
+	}
+
+	if inline.StatusCode >= 300 {
+		resp.Diagnostics.AddError(
+			"Unable to create cluster connector secret",
+			fmt.Sprintf("Unable to create cluster connector secret, got status: %s", inline.Status))
+		return
+	}
+
+	tflog.Info(ctx, "Camunda cluster connector secret created", map[string]interface{}{
+		"Name":      data.Name,
+		"ClusterId": data.ClusterId,
+	})
+
+	diags = resp.State.Set(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *CamundaClusterConnectorSecretResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data camundaClusterConnectorSecret
+
+	diags := req.State.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ctx = context.WithValue(ctx, console.ContextAccessToken, r.provider.accessToken)
+
+	secrets, _, err := r.provider.client.ClustersApi.GetSecrets(ctx, data.ClusterId.ValueString()).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Connector Secret Error",
+			fmt.Sprintf("Unable to read cluster connector secrets Name=%s, ClusterID=%s, got error: %s",
+				data.Name.ValueString(), data.ClusterId.ValueString(), err.(*console.GenericOpenAPIError).Body()),
+		)
+		return
+	}
+
+	// Find secrets name in list of secrets
+	// We also check if the value changed
+	found := false
+	for key, value := range secrets {
+		if key == data.Name.ValueString() {
+			found = true
+			if value != data.Value.ValueString() {
+				data.Value = types.StringValue(value)
+			}
```

Comment:
You can probably remove the test here, not sure it's super useful:

```suggestion
			data.Value = types.StringValue(value)
```